src/orion/core/worker/experiment.py

```diff
@@ -313,7 +313,11 @@ def configure(self, config):
         experiment._instantiate_config(self.configuration)
         experiment._instantiate_config(config)
         experiment._init_done = True
-        experiment.status = 'pending'
+
+        # If experiment is new or non-branching parameters have been changed such that the status is
+        # not 'done' anymore
+        if experiment.status is None or (experiment.status != 'broken' and not experiment.is_done):
```

Comment:
Is there a specific reason you wanted to have this in `configure`, instead of having it done in `workon`? I find it more explicit this way. Could you explain the status transition table??

Reply:
Currently the experiment was set to 'pending' by default. I restricted the change of status so that it can only become 'pending' if it is new, or if it is not broken and not completed yet.

I would totally agree to externalise the change of status. I believe this one to be the lesser evil considering the status is changed inside `Experiment.configure`.